src/core/utils.js

```diff
@@ -859,3 +861,19 @@ export function paramToValue(param, paramValues) {
 
   return values[0]
 }
+
+// adapted from https://auth0.com/docs/flows/guides/auth-code-pkce/includes/create-code-verifier
+export function generateCodeVerifier() {
```

Comment:
After realizing my code coverage for the feature was not the best, I moved the 2 methods to create the `code_verifier` and `code_challenge` in the utils file. This enables to mock properly when testing oauth2-authorize.js.